src/osx/dataview_osx.cpp

```diff
@@ -251,7 +251,14 @@ bool wxOSXDataViewModelNotifier::ValueChanged(wxDataViewItem const& item, unsign
 
 bool wxOSXDataViewModelNotifier::Cleared()
 {
-  return m_DataViewCtrlPtr->GetDataViewPeer()->Reload();
+  bool noFailureFlag;
+
+  // NOTE: See comments in ItemDeleted method above
+  m_DataViewCtrlPtr->SetDeleting(true);
+  noFailureFlag = m_DataViewCtrlPtr->GetDataViewPeer()->Reload();
+  m_DataViewCtrlPtr->SetDeleting(false);
```

Comment:
wouldn't it be better to store the former state here using `IsDeleting` and restore this state after, instead of unconditionally seting it to false

Reply:
This is consistent with the existing usage of this function. We/I can change all of them at once later (I'd also use a RAII helper instead of doing it manually and get rid of `noFailureFlag`).